core/src/main/resources/hudson/TcpSlaveAgentListener/index.jelly

```diff
@@ -34,6 +34,8 @@ THE SOFTWARE.
     if the host name is overriden, advertise that as well
   -->
   <st:header name="X-Jenkins-JNLP-Host" value="${app.tcpSlaveAgentListener.CLI_HOST_NAME}" />
+  <!-- publish the instance identity so that agents can verify match from discovery -->
+  <st:header name="X-Instance-Identity" value="${app.tcpSlaveAgentListener.identityPublicKey}" />
```

Comment:
Assuming you expect to retrieve this header from the agent, it would be advisable to have a functional test using `WebClient`.